File: services/api/dto/documentation.go
package dto

import "time"

// ── Employee ──────────────────────────────────────────────────────────────────

type DocEmployee struct {
	ID        string    `json:"id"`
	FirstName string    `json:"first_name"`
	LastName  string    `json:"last_name"`
	Role      string    `json:"role"`
	Active    bool      `json:"active"`
	CreatedAt time.Time `json:"created_at"`
}

// ── Document file ─────────────────────────────────────────────────────────────

type DocFile struct {
	ID               string    `json:"id"`
	OriginalFilename string    `json:"original_filename"`
	MimeType         string    `json:"mime_type"`
	FileSize         int64     `json:"file_size"`
	UploadedAt       time.Time `json:"uploaded_at"`
}

// ── Medical exam ──────────────────────────────────────────────────────────────

type MedicalExam struct {
	ID            string    `json:"id"`
	EmployeeID    string    `json:"employee_id"`
	CompletedDate string    `json:"completed_date"` // YYYY-MM-DD
	ExpiresAt     string    `json:"expires_at"`     // YYYY-MM-DD
	Document      *DocFile  `json:"document,omitempty"`
	CreatedAt     time.Time `json:"created_at"`
}

// ── Safety training ───────────────────────────────────────────────────────────

type SafetyRecord struct {
	ID         string     `json:"id"`
	EmployeeID string     `json:"employee_id"`
	Status     string     `json:"status"` // pending / uploaded / verified
	Document   *DocFile   `json:"document,omitempty"`
	VerifiedAt *time.Time `json:"verified_at,omitempty"`
	CreatedAt  time.Time  `json:"created_at"`
}

// ── Employment contract ───────────────────────────────────────────────────────

type EmploymentContract struct {
	ID                  string    `json:"id"`
	EmployeeID          string    `json:"employee_id"`
	ContractType        string    `json:"contract_type"` // odredeno / neodredeno
	ExpiresAt           *string   `json:"expires_at,omitempty"`
	Document            *DocFile  `json:"document,omitempty"`
	TerminatedAt        *string   `json:"terminated_at,omitempty"`
	TerminationDocument *DocFile  `json:"termination_document,omitempty"`
	CreatedAt           time.Time `json:"created_at"`
}

// ── Monthly obligation ────────────────────────────────────────────────────────

type MonthlyObligation struct {
	ID             string     `json:"id"`
	EmployeeID     string     `json:"employee_id"`
	Year           int        `json:"year"`
	Month          int        `json:"month"`
	ObligationType string     `json:"obligation_type"` // platna_lista / evidencija_radnika
	Status         string     `json:"status"`          // pending / completed
	Hours          *float64   `json:"hours,omitempty"`
	Document       *DocFile   `json:"document,omitempty"`
	CompletedAt    *time.Time `json:"completed_at,omitempty"`
	CreatedAt      time.Time  `json:"created_at"`
}

// ── Annual leave ──────────────────────────────────────────────────────────────

type AnnualLeave struct {
	ID         string    `json:"id"`
	EmployeeID string    `json:"employee_id"`
	DateFrom   *string   `json:"date_from,omitempty"`
	DateTo     *string   `json:"date_to,omitempty"`
	Document   *DocFile  `json:"document,omitempty"`
	CreatedAt  time.Time `json:"created_at"`
}

// ── Work permit ───────────────────────────────────────────────────────────────

type WorkPermit struct {
	ID         string    `json:"id"`
	EmployeeID string    `json:"employee_id"`
	EnteredAt  string    `json:"entered_at"`
	ExpiresAt  string    `json:"expires_at"`
	Document   *DocFile  `json:"document,omitempty"`
	CreatedAt  time.Time `json:"created_at"`
}

// ── Pension record ────────────────────────────────────────────────────────────

type PensionRecord struct {
	ID         string     `json:"id"`
	EmployeeID string     `json:"employee_id"`
	Status     string     `json:"status"` // pending / uploaded / verified
	Document   *DocFile   `json:"document,omitempty"`
	VerifiedAt *time.Time `json:"verified_at,omitempty"`
	CreatedAt  time.Time  `json:"created_at"`
}

// ── Health record ─────────────────────────────────────────────────────────────

type HealthRecord struct {
	ID         string     `json:"id"`
	EmployeeID string     `json:"employee_id"`
	Status     string     `json:"status"` // pending / uploaded / verified
	Document   *DocFile   `json:"document,omitempty"`
	VerifiedAt *time.Time `json:"verified_at,omitempty"`
	CreatedAt  time.Time  `json:"created_at"`
}

// ── Notification ──────────────────────────────────────────────────────────────

// DocNotificationType identifies which documentation rule triggered a notification.
const (
	DocNotifMedicalWarning   = "medical_warning"
	DocNotifMedicalUrgent    = "medical_urgent"
	DocNotifMedicalOverdue   = "medical_overdue"
	DocNotifSafetyLow        = "safety_low"
	DocNotifSafetyUrgent     = "safety_urgent"
	DocNotifContractWarning  = "contract_warning"
	DocNotifContractOverdue  = "contract_overdue"
	DocNotifPayrollWarning   = "payroll_warning"
	DocNotifPayrollOverdue   = "payroll_overdue"
	DocNotifTimesheetWarning = "timesheet_warning"
	DocNotifTimesheetOverdue = "timesheet_overdue"
	DocNotifPermitLow        = "permit_low"
	DocNotifPermitWarning    = "permit_warning"
	DocNotifPermitUrgent     = "permit_urgent"
	DocNotifPermitOverdue    = "permit_overdue"
	DocNotifPensionMissing   = "pension_missing"
	DocNotifHealthMissing    = "health_missing"
)

// DocPriority constants.
const (
	PriorityLow     = "low"
	PriorityWarning = "warning"
	PriorityUrgent  = "urgent"
	PriorityOverdue = "overdue"
)

type DocNotification struct {
	Type           string  `json:"type"`
	Priority       string  `json:"priority"`
	EmployeeID     string  `json:"employee_id"`
	EmployeeName   string  `json:"employee_name"`
	RecordID       string  `json:"record_id,omitempty"`
	Message        string  `json:"message"`
	DueDate        *string `json:"due_date,omitempty"`
	DaysRemaining  *int    `json:"days_remaining,omitempty"`
	DaysOverdue    *int    `json:"days_overdue,omitempty"`
	Year           *int    `json:"year,omitempty"`
	Month          *int    `json:"month,omitempty"`
	ObligationType *string `json:"obligation_type,omitempty"`
}

// ── Employee detail (summary) ─────────────────────────────────────────────────

type EmployeeDocSummary struct {
	Employee       DocEmployee         `json:"employee"`
	LatestMedical  *MedicalExam        `json:"latest_medical,omitempty"`
	SafetyRecord   *SafetyRecord       `json:"safety_record,omitempty"`
	ActiveContract *EmploymentContract `json:"active_contract,omitempty"`
	LatestPermit   *WorkPermit         `json:"latest_permit,omitempty"`
	PensionRecord  *PensionRecord      `json:"pension_record,omitempty"`
	HealthRecord   *HealthRecord       `json:"health_record,omitempty"`
}

File: apps/web/lib/types/documentation.ts
export interface DocEmployee {
  id: string
  first_name: string
  last_name: string
  role: string
  active: boolean
  created_at: string
}

export interface DocFile {
  id: string
  original_filename: string
  mime_type: string
  file_size: number
  uploaded_at: string
}

export interface MedicalExam {
  id: string
  employee_id: string
  completed_date: string
  expires_at: string
  document?: DocFile
  created_at: string
}

export interface SafetyRecord {
  id: string
  employee_id: string
  status: 'pending' | 'uploaded' | 'verified'
  document?: DocFile
  verified_at?: string
  created_at: string
}

export interface EmploymentContract {
  id: string
  employee_id: string
  contract_type: 'odredeno' | 'neodredeno'
  expires_at?: string
  document?: DocFile
  terminated_at?: string
  termination_document?: DocFile
  created_at: string
}

export interface MonthlyObligation {
  id: string
  employee_id: string
  year: number
  month: number
  obligation_type: 'platna_lista' | 'evidencija_radnika'
  status: 'pending' | 'completed'
  hours?: number
  document?: DocFile
  completed_at?: string
  created_at: string
}

export interface AnnualLeave {
  id: string
  employee_id: string
  date_from?: string
  date_to?: string
  document?: DocFile
  created_at: string
}

export interface WorkPermit {
  id: string
  employee_id: string
  entered_at: string
  expires_at: string
  document?: DocFile
  created_at: string
}

export interface PensionRecord {
  id: string
  employee_id: string
  status: 'pending' | 'uploaded' | 'verified'
  document?: DocFile
  verified_at?: string
  created_at: string
}

export interface HealthRecord {
  id: string
  employee_id: string
  status: 'pending' | 'uploaded' | 'verified'
  document?: DocFile
  verified_at?: string
  created_at: string
}

export interface DocNotification {
  type: string
  priority: 'low' | 'warning' | 'urgent' | 'overdue'
  employee_id: string
  employee_name: string
  record_id?: string
  message: string
  due_date?: string
  days_remaining?: number
  days_overdue?: number
  year?: number
  month?: number
  obligation_type?: string
}

export interface EmployeeDocSummary {
  employee: DocEmployee
  latest_medical?: MedicalExam
  safety_record?: SafetyRecord
  active_contract?: EmploymentContract
  latest_permit?: WorkPermit
  pension_record?: PensionRecord
  health_record?: HealthRecord
}
